### PR TITLE
Editor mobile devices preview mode

### PR DESCRIPTION
**Pull request by gorgeousvlad**

### CODE REVIEW

File: src/editor/components/DeviceEmulation/DeviceEmulation.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    useEffect(() => {
+        if (deviceIframeRef.current && initialData) {
+            deviceIframeRef.current.onDataUpdate(initialData);
+        }
+    }, [initialData]);
+
+    return <div className={b({active, device})} ref={containerRef} />;
+};
+
+const DeviceEmulation = ({children, mode}: DeviceEmulationProps) => {
```
Comment (NikitaCG) [inline src/editor/components/DeviceEmulation/DeviceEmulation.tsx:66]: return immediately

File: src/editor/components/DeviceEmulation/DeviceEmulation.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+const mobileDevices = [ViewModeItem.Tablet, ViewModeItem.Mobile] as const;
+const isMobileDevice = (mode: ViewModeItem): mode is MobileDevice =>
+    mobileDevices.includes(mode as MobileDevice);
+
+interface DeviceEmulationMobileProps extends PropsWithChildren {
+    device: MobileDevice;
+    active: boolean;
+}
+
+const DeviceEmulationMobile = ({device, active}: DeviceEmulationMobileProps) => {
```
Comment (NikitaCG) [inline src/editor/components/DeviceEmulation/DeviceEmulation.tsx:27]: maybe split into two files? to make it easier for developers to navigate 
```
DeviceEmulationMobile.tsx 
DeviceEmulation.tsx
```

File: src/editor/components/DeviceEmulation/DeviceEmulation.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        }
+    }, [initialData]);
+
+    return <div className={b({active, device})} ref={containerRef} />;
+};
+
+const DeviceEmulation = ({children, mode}: DeviceEmulationProps) => {
+    return (
+        <Fragment>
+            {!isMobileDevice(mode) && children}
```
Comment (NikitaCG) [inline src/editor/components/DeviceEmulation/DeviceEmulation.tsx:69]: we can use ternary operator in this render method ?
```
!isMobileDevice(mode) 
  ? children 
  : (
    <DeviceEmulationMobile key={device} device={device} active={mode === device}>
                    {children}
     </DeviceEmulationMobile>
  )
```

**Review by NikitaCG — CHANGES_REQUESTED** [review]

**Review by niktverd — APPROVED** [review]

File: src/editor/components/DeviceEmulation/DeviceEmulation.tsx (hunk 3 above)
Comment (gorgeousvlad) [reply src/editor/components/DeviceEmulation/DeviceEmulation.tsx:69]: no, the idea is to render it all at same time and then hide, not to create iframe window by every render

**Review by NikitaCG — APPROVED** [review]

**Review by yuberdysheva — APPROVED** [review]

**Pull request merged**